pyext/src/rmsd_calculation.py

```diff
@@ -89,6 +89,20 @@ def get_rmfs_coordinates(path, idfile_A, idfile_B, subunit_name):
 
             if subunit_name:
                 s0 = IMP.atom.Selection(h, resolution=1, molecule=subunit_name)
+            elif selection:
+                s0=None
+                for idx,selected_range in enumerate(selection.values()):
+                    if idx==0:
+                        s0 = IMP.atom.Selection(h, resolution=1,
+                                molecule=str(selected_range[0][2]), 
+                                residue_indexes=range(selected_range[0][0], 
+                                selected_range[0][1]))
+                    else:
+                        s1 = IMP.atom.Selection(h, resolution=1,
+                                molecule=str(selected_range[0][2]), 
+                                residue_indexes=range(selected_range[0][0], 
+                                selected_range[0][1]))
```

Comment:
Same `Selection` object is created in each branch of the `if`. Create it before the `if` instead so that you don't duplicate code.

Reply:
Sorry...I am not familiar enough with the imp atom selection. I just wanted to create a selection that expands with the | operator as the loop progresses. So i thought I had to do a selection first (s0 when it's the first key of the dictionary) and then updating it by adding s1 at each iteration after the first... I guess not?

Reply:
Nothing to do with IMP specifically here, but you could reduce code duplication with something like

```
s = IMP.atom.Selection(h, resolution=1,
                       molecule=str(selected_range[0][2]), 
                       residue_indexes=range(selected_range[0][0], 
                       selected_range[0][1]))
if idx == 0:
    s0 = s
else:
    s0 |= s
```

Reply:
thanks for the tip! Anyways I think it's done on my end...